src/odemis/driver/andorcam2.py

```diff
@@ -1884,6 +1888,25 @@ def _getMaxBPP(self):
         assert(mbpp > 0)
         return mbpp
 
+    def _update_settings_if_not_acquiring(self):
+        """
+        Try to directly configure the camera, if it's not acquiring.
+        This way the .frameDuration value is updated
+        """
+        # See if the camera is acquiring: has hw_lock and status is acquiring
+        # If we fail to acquire the lock, it means the acquisition is ongoing
+        # so no need/not possible to update the settings. They will be updated next frame.
+        if self.hw_lock.acquire(timeout=0.1):
+            try:
+                if self.GetStatus() != AndorV2DLL.DRV_ACQUIRING:
+                    self._update_settings()
+            finally:
+                self.hw_lock.release()
+        # TODO: might still need to update the settings at the end of the acquisition, in case
+        # the VAs were just changed before ending the acquisition?
+        # TODO: or should we just try to guess? (ie, duplicate the behaviour of GetAcquisitionTimings())
+        # We could also try to cache the result of GetAcquisitionTimings() for such cases.
```

Comment:
what would be the advantage of guessing?

Reply:
It would provide some feedback to the user about which value is acceptable. Still, as it's a guess, we could pick a wrong one, which would be quite annoying, as the behaviour would be different whether the camera is playing or not. => Let's not do that. I've updated the comment.